pkg/dynamicinstrumentation/ditypes/config.go

```diff
@@ -0,0 +1,337 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+// Package ditypes contains various datatypes and otherwise shared components
+// used by all the packages in dynamic instrumentation
+package ditypes
+
+import (
+	"debug/dwarf"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ratelimiter"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	"github.com/google/uuid"
+)
+
+const ConfigBPFProbeID = "config" // ConfigBPFProbeID is the ID used for the config bpf program
+
+var (
+	CaptureParameters       = true  // CaptureParameters is the default value for if probes should capture parameter values
+	ArgumentsMaxSize        = 10000 // ArgumentsMaxSize is the default size in bytes of the output buffer used for param values
+	StringMaxSize           = 512   // StringMaxSize is the default size in bytes of a single string
+	MaxReferenceDepth uint8 = 4     // MaxReferenceDepth is the default depth that DI will traverse datatypes for capturing values
+	MaxFieldCount           = 20    // MaxFieldCount is the default limit for how many fields DI will capture in a single data type
+	SliceMaxSize            = 1800  // SliceMaxSize is the default limit in bytes of a slice
+	SliceMaxLength          = 100   // SliceMaxLength is the default limit in number of elements of a slice
```

Comment:
Suggestion: Consider having limits used in eBPF to be power of twos. They help in constraining the verifier without branches.